src/main/java/org/springframework/data/elasticsearch/core/query/Query.java

```diff
@@ -229,4 +229,29 @@ default Integer getMaxResults() {
 	 * @since 4.0
 	 */
 	boolean getTrackTotalHits();
+
+	/**
+	 * For queries that are used in delete request, these are internally handled by Elasticsearch as scroll/bulk delete queries.
+	 * 
+	 * @return the scrolltime settings
+	 * @since 4.0
+	 */
+	@Nullable
+	Long getScrollTimeInMillis();
+
+	/**
+	 * For queries that are used in delete request, these are internally handled by Elasticsearch as scroll/bulk delete queries.
+	 * 
+	 * @param scrollTimeInMillis the scrolltime settings
+	 * @since 4.0
+	 */
+	void setScrollTimeInMillis(Long scrollTimeInMillis);
```

Comment:
How about changing the base property to `ScrollTime(Duration)`? `inMillis` is a quite dated pattern before we had proper duration types.